### PR TITLE
ignore group membership

### PR DESCRIPTION
**Pull request by navisatebi**
Created a new branch, to keep commits clean. Formatting and get Task changed as per requirements.
IT added to cover the new set of requirements.

### CODE REVIEW

File: src/main/java/uk/ac/ebi/tsc/tesk/util/component/KubernetesClientWrapper.java [hunk 1]
```diff
@@ -94,6 +98,9 @@ public V1JobList listAllTaskmasterJobsForUser(String pageToken, Integer itemsPer
             // and optionally also to only those jobs, which were created bu the user
             labelSelector += "," + user.getLabelSelector();
         }
+        if(authorisationProperties.isIgnoreGroupMembership()) {
+            labelSelector += "," + new StringJoiner("=").add(LABEL_USERID_KEY).add(user.getUsername()).toString();
```
Comment (aniewielska) [inline src/main/java/uk/ac/ebi/tsc/tesk/util/component/KubernetesClientWrapper.java:102]: Could you please follow the formatting used in the code (I asked about it before)?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -22,17 +22,17 @@ tesk.api.service-info.documentation=https://github.com/EMBL-EBI-TSI/TESK
...
 tesk.api.taskmaster.ftp.secret-name=
-tesk.api.taskmaster.service-account-name=default
+tesk.api.taskmaster.service-account-name=taskmaster
 tesk.api.taskmaster.debug=false
 
 tesk.api.taskmaster.executor-secret.name=
 tesk.api.taskmaster.executor-secret.mount-path=/secret
 
-spring.profiles.active=noauth
+spring.profiles.active=auth
```
Comment (aniewielska) [inline src/main/resources/application.properties:35]: Could you please not change the default to authenticated mode?

File: src/main/java/uk/ac/ebi/tsc/tesk/service/TesService.java [hunk 3]
```diff
@@ -21,9 +21,9 @@
      * Creates new TES task, by converting input and calling create method.
      * In case of detecting duplicated task ID, retries with new generated ID (up to a limit of retries)
      */
-    @PreAuthorize("hasRole(@authorisationProperties.baseGroupFull) AND" +
+    @PreAuthorize("(authentication.authenticated AND @authorisationProperties.isIgnoreGroupMembership()) OR (hasRole(@authorisationProperties.baseGroupFull) AND" +
```
Comment (aniewielska) [inline src/main/java/uk/ac/ebi/tsc/tesk/service/TesService.java:24]: Is `authentication.authenticated` really necessary? (I believe it is not).

File: src/main/java/uk/ac/ebi/tsc/tesk/service/TesService.java [hunk 4]
```diff
@@ -34,7 +34,7 @@
      * @param view   - one of {@link TaskView} values, decides on how much detail is put in results
      * @return - TES task details
      */
-    @PostAuthorize("(#user.username == returnObject.logs[0].metadata['USER_ID'] AND returnObject.logs[0].metadata['GROUP_NAME'] != null AND #user.isGroupMember(returnObject.logs[0].metadata['GROUP_NAME']))" +
+    @PostAuthorize("(authentication.authenticated AND @authorisationProperties.isIgnoreGroupMembership() AND #user.username == returnObject.logs[0].metadata['USER_ID']) OR (#user.username == returnObject.logs[0].metadata['USER_ID'] AND returnObject.logs[0].metadata['GROUP_NAME'] != null AND #user.isGroupMember(returnObject.logs[0].metadata['GROUP_NAME']))" +
             " OR (#user.teskAdmin)" +
```
Comment (aniewielska) [inline src/main/java/uk/ac/ebi/tsc/tesk/service/TesService.java:38]: This condition can be simplified: isOwner and either ignoreMembership or belong to a group.

File: src/integration-test/java/uk/ac/ebi/tsc/tesk/AuthIT.java [hunk 5]
```diff
@@ -38,7 +38,7 @@
 @AutoConfigureMockMvc
 @TestPropertySource(locations = {"classpath:application.properties"},
         properties = {"security.oauth2.resource.user-info-uri = http://localhost:8090",
-                "spring.profiles.active=auth"})
+                "spring.profiles.active=auth","tesk.api.authorisation.ignoreGroupMembership=false"})
```
Comment (aniewielska) [inline src/integration-test/java/uk/ac/ebi/tsc/tesk/AuthIT.java:41]: Why is this additional property needed here? It should work by default.

File: src/integration-test/java/uk/ac/ebi/tsc/tesk/NoAuthIT.java [hunk 6]
```diff
@@ -37,7 +37,7 @@
 @AutoConfigureMockMvc
 @TestPropertySource(locations = {"classpath:application.properties"},
         properties = {"security.oauth2.resource.user-info-uri = http://localhost:8090",
-                "spring.profiles.active=noauth"})
+                "spring.profiles.active=noauth","tesk.api.authorisation.ignoreGroupMembership=false"})
```
Comment (aniewielska) [inline src/integration-test/java/uk/ac/ebi/tsc/tesk/NoAuthIT.java:40]: Same as above.

File: src/main/java/uk/ac/ebi/tsc/tesk/service/TesService.java [hunk 7]
```diff
@@ -21,9 +21,9 @@
      * Creates new TES task, by converting input and calling create method.
      * In case of detecting duplicated task ID, retries with new generated ID (up to a limit of retries)
      */
-    @PreAuthorize("hasRole(@authorisationProperties.baseGroupFull) AND" +
+    @PreAuthorize("(authentication.authenticated AND @authorisationProperties.isIgnoreGroupMembership()) OR (hasRole(@authorisationProperties.baseGroupFull) AND" +
             "(#user.member OR #user.teskAdmin) AND" +
```
Comment (aniewielska) [inline src/main/java/uk/ac/ebi/tsc/tesk/service/TesService.java:25]: Is `authentication.authenticated` necessary? What happens, if you remove it?

File: src/main/java/uk/ac/ebi/tsc/tesk/service/TesService.java [hunk 8]
```diff
@@ -34,7 +34,7 @@
      * @param view   - one of {@link TaskView} values, decides on how much detail is put in results
      * @return - TES task details
      */
-    @PostAuthorize("(#user.username == returnObject.logs[0].metadata['USER_ID'] AND returnObject.logs[0].metadata['GROUP_NAME'] != null AND #user.isGroupMember(returnObject.logs[0].metadata['GROUP_NAME']))" +
+    @PostAuthorize("(authentication.authenticated AND @authorisationProperties.isIgnoreGroupMembership() AND #user.username == returnObject.logs[0].metadata['USER_ID']) OR (#user.username == returnObject.logs[0].metadata['USER_ID'] AND returnObject.logs[0].metadata['GROUP_NAME'] != null AND #user.isGroupMember(returnObject.logs[0].metadata['GROUP_NAME']))" +
```
Comment (aniewielska) [inline src/main/java/uk/ac/ebi/tsc/tesk/service/TesService.java:37]: This condition can be simplified to: isOwner and (is a groupMember or ignoreMembership)

File: src/main/java/uk/ac/ebi/tsc/tesk/util/component/KubernetesClientWrapper.java [hunk 9]
```diff
@@ -94,6 +98,9 @@ public V1JobList listAllTaskmasterJobsForUser(String pageToken, Integer itemsPer
...
+        }
```
Comment (aniewielska) [inline src/main/java/uk/ac/ebi/tsc/tesk/util/component/KubernetesClientWrapper.java:103]: Here formatting does not follow the standards used in the code (space after if) and I believe this code does not work for a user, who belongs to any group: the tasks outside of any groups, but still belonging to the user will not show up. The idea is that the label selector for `ignoreMembership` should not include checking groups at all.

File: src/main/resources/application.properties (hunk 2 above)
Comment (aniewielska) [inline src/main/resources/application.properties:35]: Please do not set authenticated mode by default.

File: src/integration-test/java/uk/ac/ebi/tsc/tesk/AuthIgnoreGroupMembershipIT.java [hunk 10]
```diff
@@ -0,0 +1,533 @@
...
+    static class KubernetesClientMock {
+        @Bean
+        @Primary
+        public ApiClient kubernetesApiClient() {
+
+            return Config.fromUrl("http://localhost:9000", false);
+
+        }
+    }
+
```
Comment (aniewielska) [inline src/integration-test/java/uk/ac/ebi/tsc/tesk/AuthIgnoreGroupMembershipIT.java:62]: I think we can show the point of membership not interfering with ignoreMembership and still not copy all the tests from the previous test suite. Are they really necessary?

File: src/integration-test/java/uk/ac/ebi/tsc/tesk/AuthIgnoreGroupMembershipIT.java [hunk 11]
```diff
@@ -0,0 +1,533 @@
...
+    }
+
+    @Test
+    public void nonmember_getList() throws Exception {
+
+        mockElixir.givenThat(
+                WireMock.get("/")
+                        .willReturn(okJson("{\"sub\" : \"123\",  \"eduperson_entitlement\" : [\"sth\",\"urn:geant:elixir-europe.org:group:elixir:GA4GH:GA4GH-CAP#perun.elixir-czech.cz\"]}")));
+
```
Comment (aniewielska) [inline src/integration-test/java/uk/ac/ebi/tsc/tesk/AuthIgnoreGroupMembershipIT.java:494]: Doesn't the non-member belong here to the same group as member (in fact being a member)?

**Review by aniewielska — CHANGES_REQUESTED** [review]

**Comment by navisatebi** [comment]
More development in another branch

**Pull request closed without merging**